app/Platform/Commands/SyncAchievementAuthors.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Commands;
+
+use App\Models\Achievement;
+use App\Models\User;
+use App\Platform\Enums\AchievementAuthorTask;
+use Illuminate\Console\Command;
+
+class SyncAchievementAuthors extends Command
+{
+    protected $signature = 'ra:sync:achievement-authors
+                            {achievementId? : Target achievement ID. Leave empty to sync from all achievements.}';
+    protected $description = "Write all LOGIC authorship data to achievement_authors";
+
+    public function __construct(
+    ) {
+        parent::__construct();
+    }
+
+    public function handle(): void
+    {
+        $achievementId = $this->argument('achievementId');
+
+        if ($achievementId) {
+            $achievement = Achievement::findOrFail($achievementId);
+
+            $this->syncAchievementLogicAuthors($achievement);
+        } else {
+            $distinctAchievementIdsCount = Achievement::count();
+
+            $this->info("\nUpserting logic authorship credit for {$distinctAchievementIdsCount} achievements.");
+
+            $progressBar = $this->output->createProgressBar($distinctAchievementIdsCount);
+            Achievement::query()->chunk(100, function ($achievements) use ($progressBar) {
+                foreach ($achievements as $achievement) {
+                    $this->syncAchievementLogicAuthors($achievement, quiet: true);
+
+                    $progressBar->advance();
+                }
+            });
+            $progressBar->finish();
+
+            $this->info("\nCompleted upserting logic authorship credit for {$distinctAchievementIdsCount} achievements.");
+        }
+    }
+
+    private function syncAchievementLogicAuthors(Achievement $achievement, bool $quiet = false): void
+    {
+        if (!$quiet) {
+            $this->info("\nUpserting logic authorship credit for achievement [{$achievement->id}:{$achievement->title}] ({$achievement->game->title}).");
+        }
+
+        $addedAuthors = 0;
+
+        $achievementAuthor = $achievement->developer()->withTrashed()->first();
```

Comment:
I'm not certain the `withTrashed` is working properly: http://localhost:64000/manage/achievements/44672

Reply:
Good catch, this is now fixed in latest.

![Screenshot 2024-09-16 at 3 55 24 PM](https://github.com/user-attachments/assets/944f6daf-525d-49d3-948e-10f9e61ea16c)

Reply:
For non-moderators, navigating to that deleted user's page is going to lead to a forbidden access error. I'm not sure if this is worth worrying about, given the context is Filament.